### PR TITLE
Set up TypeScript

### PR DESCRIPTION
**Pull request by ZahraTee**
Remove the PropTypes enforcement in ESLint and introduce TypeScript instead.

TypeScript will be automatically transpiled by webpack using ts-loader.
Basic configuration added and a few files (breadcrumbs, err_code, etc) minimally converted.

### CODE REVIEW

File: static/breadCrumbs.tsx [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+interface IBreadCrumbProps {
+    getSource: (path: string, lineStart?: number) => void,
+    path: string
+}
+
+interface IBreadCrumbState {
+    
+}
+
+export default class BreadCrumbs extends React.Component<IBreadCrumbProps, IBreadCrumbState> {
```
Comment (nrc) [inline static/breadCrumbs.tsx]: Does using TypeScript mean we can't use the function version of React components?

Comment (ZahraTee) [reply static/breadCrumbs.tsx]: No, I'll change it back.

**Comment by ZahraTee** [comment]
@nrc updated!

**Comment by nrc** [comment]
Thanks!

**Pull request merged**